src/main/kotlin/com/openlattice/users/Auth0SyncTask.kt

```diff
@@ -69,25 +64,25 @@ class Auth0SyncTask : HazelcastFixedRateTask<Auth0SyncTaskDependencies>, Hazelca
     override fun runTask() {
         val ds = getDependency()
         logger.info("Synchronizing users.")
-        try {
-            ds.userListingService
-                    .getUsers()
-                    .map {
-                        syncSemaphore.acquire()
-                        ds.executor.submit { ds.users.syncUser(it) }
-                    }
-                    .forEach {
-                        it.get()
-                        syncSemaphore.release()
+        ds.userListingService
+                .getUsers()
+                .map {
+                    syncSemaphore.acquire()
+                    ds.executor.submit {
+                        try {
+                            ds.users.syncUser(it)
+                        } catch (ex: Exception) {
+                            logger.error("Unable to synchronize user", ex)
+                        } finally {
+                            syncSemaphore.release()
+                        }
                     }
-
-        } catch (ex: Exception) {
-            logger.error("Unable to synchronize users", ex)
-            return
-        }
-
+                }
+                // we want to materialize the list of futures so the work happens in the background.
+                .toList()
+                .forEach {
+                    it.get()
```

Comment:
this call is now safe(ish?), it should only throw on non-syncUser problems like interruptedexception.